### DISCUSSION TITLE
Are string literals in method arguments considered as LOC ?

### DISCUSSION THREAD

**Discussion by nanafan93** [detekt/detekt#6404] [Q&A] [answered]
String literals as method parameters seem to trigger the **LongMethod** rule ? 
1. In my example below, I am passing a JSON string to my assertion method and that fails the build when I run detekt.
3. When I extract the string into a variable and pass the variable, it seems to work fine.
4. Is this intentional and correct behaviour ? I wouldn't think that string literals are counted as LOC. Is there any way to change the `threshold` only for a particular file/pattern of files ? 

```
   @Test
    fun doSomething(@Autowired mvc: MockMvc) {
        val jwt = createJwt(
            customerCode = CUSTOMER_CODE,
            privateKeyLocation = "somePath"
        )
        val request = MockMvcRequestBuilders.get("/v1/blahblah")
            .accept(MediaType.APPLICATION_JSON)
            .contentType(MediaType.APPLICATION_JSON)
            .header("Authorization", "Bearer $jwt")
            .header("X-Header-Code", CUSTOMER_CODE)

        mvc.perform(request)
            .andExpect(MockMvcResultMatchers.status().isOk)
            .andExpect(MockMvcResultMatchers.content().contentType(MediaType.APPLICATION_JSON))
            .andExpect(
                MockMvcResultMatchers.content().json(
                    """
        Intentional blank lines            
                        
{
  "filters": [
    {
      "filterType": "MULTI_SELECT",
      "label": "idc",
      "queryParam": "hehe",
      "placeHolder": "Haha...",
      "options": [
        {
          "label": "Small",
          "value": "6"
        }
      ]
    },
    {
      "filterType": "MULTI_SELECT",
      "label": "Blah",
      "queryParam": "boo",
      "placeHolder": "floor...",
      "options": [
        {
          "label": "Big",
          "value": "1"
        }
      ]
    },
    {
      "filterType": "MULTI_SELECT",
      "label": "kek",
      "queryParam": "lol",
      "placeHolder": "lul...",
      "options": [
        {
          "label": "2008",
          "value": "1"
        }
      ]
    }
  ]
}""",
                    false
                )
            )
    }
}
```

**Comment by cortinico** [comment, marked as answer]
> String literals as method parameters seem to trigger the LongMethod rule ?

Yes they do, we just check the lines of code size of a method to trigger that rule.

As for your specific example, you probably want to either:
1. Move this string to a variable
2. Suppress `LongMethod` for tests using an `exclude` directive in your config file
3. Suppress `LongMethod` for tests using the `ignoreAnnotated` suppressors (see here https://detekt.dev/docs/next/introduction/suppressors/#annotation-suppressor)
